CONTRIBUTING.rst

```diff
@@ -3,45 +3,98 @@
-issues.
-See `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ to get
-started.
+issues such as `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ ,
+`known bugs <https://github.com/nilearn/nilearn/labels/Bug>`_ or
+`proposed enhancements <https://github.com/nilearn/nilearn/labels/Enhancement>`_.
 
 If an issue does not already exist for a potential contribution, we ask that
 you first open an `issue <https://github.com/nilearn/nilearn/issues>`_ before
-sending a :ref:`pull request`.
+sending a :ref:`pull request` to discuss in advance scope and potential design
+choices.
+
+
+Contribution Guidelines
+---------------------------
+
+We ask every contributor to respect common guidelines. Those are inspired from
+`scikit-learn
+<https://scikit-learn.org/stable/developers/contributing.html#contributing-code>`_
+and ensure Nilearn remains simple to understand, efficient and maintanable.
+For example tests need to run quickly in order not to burden the development process.
+To keep continuous integration efficient with our limited infrastructure, running
+all the examples must lead to downloading a limited amount of data (gigabytes)
+and execute in a reasonable amount of time (a few hours).
+
+Those guidelines will hence be checked during reviewing process.
+
+
++--------------+-------------+---------------------------------------+
+|              | Which PR ?  |        Guidelines                     |
++==============+=============+=======================================+
+|              |             | - Clearly showcase benefits           |
+|  Examples    | New features| - Run in a few minutes                |
+|              |             | - Use light data from Nilearn         |
+|              |             | - Renders well after build            |
++--------------+-------------+---------------------------------------+
+|              |             | - Test type is adapted to behavior    |
+|              |             | - Tests pass continuous integration   |
+|              |  Bugfixes   | - Doesn't decrease coverage           |
+|    Tests     | New features| - Fast, using small mocked data       |
+|              |             | - Atomic (one per function) and seeded|
+|              |             | - For Bugfixes: non-regression test   |
++--------------+-------------+---------------------------------------+
+|              |             | - Variables, functions, arguments     |
+|              |             | have clear and consistent names       |
+|              |             | - Easy to read, PEP8                  |
+| Coding Style |    Any      | - Clear docstring of public functions |
+|              |             | - Low redundancy                      |
+|              |             | - No new dependency                   |
+|              |             | - Backward compatibility              |
```

Comment:
Would it be reasonable to have a section on coding style? If not, then I'd probably just clarify that the docstring should be in `numpydoc` format.

Reply:
I think it would yes.

Reply:
even if a section on coding style is added I would +1 @tsalo 's suggestion of mentioning numpydoc here

Reply:
not sure where it should go but maybe we could say somewhere that the pr should implement only one thing (what has been discussed in the issue) and not make changes that are not necessary for that single goal -- avoid "unrelated changes"

Reply:
do you have a clear ref for numpydoc ? I'm not sure to where I should point

Reply:
https://numpydoc.readthedocs.io/en/latest/format.html